runtime/parachains/src/initializer.rs

```diff
@@ -98,12 +96,12 @@ decl_storage! {
 		HasInitialized: Option<()>;
 		/// Buffered session changes along with the block number at which they should be applied.
 		///
-		/// Typically this will be empty or one element long, with the single element having a block
-		/// number of the next block.
+		/// Typically this will be empty or one element long. Apart from that this item never hits
+		/// the storage.
```

Comment:
The comment continues below that `BufferedSessionChange` is a `Vec` to handle edge cases. Maybe I haven't understood yet all the implications of the changes made here, but if we know that this is always exactly 0 or 1 element long, then we should probably change it to an `Option`. Otherwise, this comment shouldn't contradict itself.

Reply:
Now that you say this I am not sure what kind of edge-cases `Vec<_>` would save us from (both in the current and previous versions)

@rphmeier ?

Reply:
The edge case we were working with before was:
  1. We receive a buffered session change after `on_finalize`
  2. There is then a runtime upgrade or governance intervention at the next block which causes us to receive a new runtime upgrade before on-initialize. Thus we'd have 2 session changes pending.

I think a similar situation could still arise if runtime upgrades toy with the moment session changes are issued.

@coriolinus . note that it says 'typically', so there's no contradiction. If you go back to the original PR where this was introduced, I think you'd find an almost exact replica of this review comment. 

Although we won't receive session change notifications after `on_finalize` in practice, any that are received after `on_finalize` will actually hit the storage and then will be applied in `on_finalize` of the next block. that shouldn't break any of our invariants around predicting the child session, though.